packages/editor/src/components/provider/disable-non-page-content-blocks.js

```diff
@@ -21,6 +21,14 @@ export default function DisableNonPageContentBlocks() {
 	const contentOnlyIds = useSelect( ( select ) => {
 		const { getBlocksByName, getBlockParents, getBlockName } =
 			select( blockEditorStore );
+
+		const canEditTemplate =
+			select( coreStore ).canUser( 'create', 'templates' ) ?? false;
+
+		if ( canEditTemplate ) {
+			CONTENT_ONLY_BLOCKS.push( 'core/template-part' );
+		}
+
```

Comment:
I'm not entirely sur, we should change this to be honest. Why disable template parts because of the "rights". Any particular reason for this change. I mean sure the user can't do anything with the block but I feel for the "rights" it should be handled directly by the block like it's already the case.

Reply:
@youknowriad Yeah whilst testing the PR before I made this change I noticed the following: 


https://github.com/WordPress/gutenberg/assets/20684594/f93780aa-e5d1-48f6-b7fd-2c5d84cbe211


I found it odd that the template parts that are disabled/non-editable are still selectable / show their purple outlines. So I added this logic here and now it looks/works like this:


https://github.com/WordPress/gutenberg/assets/20684594/23013652-6ab3-401b-a067-e155bda1b238

Reply:
Personally, I find it ok to be able to select the template part even if you can't edit it. I believe it will have the exact same behavior if the template part is within post content for instance.

Reply:
I'm fine removing it here to reduce the scope of the PR.

> I believe it will have the exact same behavior if the template part is within post content for instance.

Currently it is not supported to insert template parts into post content though

Reply:
I'm fine either way 🤷

Reply:
I'd split this out into a separate PR and ask for design feedback.

Reply:
I moved this change into https://github.com/WordPress/gutenberg/pull/60812 :)